x-pack/plugins/search_playground/server/lib/fetch_query_source_fields.ts

```diff
@@ -21,56 +61,51 @@ export const fetchFields = async (
     index: indices,
   });
 
-  const indexDocs = [];
+  const modelIdFields = getModelIdFields(fieldCapabilities);
 
-  for (const index of indices) {
-    const x = await client.asCurrentUser.search({
+  const indicesAggs = await Promise.all(
+    indices.map(async (index) => ({
       index,
-      body: {
-        query: {
-          match_all: {},
-        },
-        size: 1,
-      },
-    });
-
-    if (x.hits.total !== 0) {
-      indexDocs.push({
+      doc: await client.asCurrentUser.search({
         index,
-        doc: x.hits.hits[0],
-      });
-    }
-  }
+        body: {
+          size: 0,
+          aggs: modelIdFields.reduce(
```

Comment:
approach makes sense to me - I was thinking whether there's any way we can improve this query since we are not interested in the aggregation doc count, but rather just the bucket values,  but I cannot think of a different approach.
was thinking that maybe we can include a top level boolean query to include exists queries on the model_id fiels, but I don't think it's any better or worth the optimization.

Reply:
yeh its a good consideration. The other part to this (which will be addressed in another issue) is that we want to also detect when there are more than one model present in the field and the query will fail. With a term size of 2, we can detect this condition and inform the user.